app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt

```diff
@@ -1416,4 +1421,177 @@ class PatientRepositoryAndroidTest {
     assertThat(patientRepository.findPatientWithBusinessId(deletedBusinessIdentifier).blockingFirst()).isEqualTo(None)
     assertThat(patientRepository.findPatientWithBusinessId("missing_identifier").blockingFirst()).isEqualTo(None)
   }
+  
+  @Test
+  fun if_patient_is_defaulter_then_isPatientDefaulter_should_return_true() {
+
+    val defaulterPatients = arrayListOf<String>()
+
+    fun savePatientRecord(
+        fullName: String,
+        bpMeasurement: List<BloodPressureMeasurement>?,
+        hasHadHeartAttack: MedicalHistory.Answer = NO,
+        hasHadStroke: MedicalHistory.Answer = NO,
+        hasDiabetes: MedicalHistory.Answer = NO,
+        hasHadKidneyDisease: MedicalHistory.Answer = NO,
+        protocolDrug: ProtocolDrug?,
+        appointmentDate: LocalDate?
+    ) {
+      val patientUuid = patientRepository.saveOngoingEntry(testData.ongoingPatientEntry(fullName = fullName))
+          .andThen(patientRepository.saveOngoingEntryAsPatient())
+          .blockingGet()
+          .uuid
+
+      bpMeasurement?.forEach {
+        bloodPressureRepository.saveMeasurement(patientUuid, it.systolic, it.diastolic, it.createdAt).blockingGet()
+      }
+      medicalHistoryRepository.save(listOf(testData.medicalHistory(
+          patientUuid = patientUuid,
+          hasDiabetes = hasDiabetes,
+          hasHadHeartAttack = hasHadHeartAttack,
+          hasHadKidneyDisease = hasHadKidneyDisease,
+          hasHadStroke = hasHadStroke))).blockingAwait()
+
+      protocolDrug?.let {
+        prescriptionRepository.savePrescription(patientUuid = patientUuid, drug = it).blockingAwait()
+      }
+
+      appointmentDate?.let {
+        appointmentRepository.schedule(patientUuid = patientUuid, appointmentDate = it).blockingGet()
+      }
+
+      val isDefaulter = patientRepository.isPatientDefaulter(patientUuid).blockingFirst()
```

Comment:
This test is also not readable since the test setup (saving the patient) is mixed with running the actual method that is being tested (`PatientRepository#isPatientADefaulter`).

In general, most tests should follow a structure like

```kotlin
fun test() {
  fun savePatient() {}

  // given
 val uuid1 = savePatientRecord()
 val uuid2 = savePatientRecord()
 ..

 // when
 val isPatient1ADefaulter = patientRepository.isPatientDefaulter(uuid1)
 val isPatient2ADefaulter = patientRepository.isPatientDefauler(uuid2)
 ..

 // expect
 assertThat(isPatient1ADefaulter.isTrue()
 assertThat(isPatient2ADefaulter).isFalse()
}
```

Following this rule makes the test a lot more readable and gives it a logical flow.